### ISSUE TITLE
Failed to recognise Laravels Macro functions

### ISSUE THREAD

**Issue by Nizari** [nunomaduro/larastan#563]
- Laravel Version: 7.10.2
- Larastan Version: 0.5.7
- `--level 5` used: (but already breaks on `--level 2`

### Description

Since upgrading to `v0.5.5` => `v0.5.7` it seems it doesn't recognise Laravels `Macro` functionality and returns the below error.

```console
 ------ ------------------------------------------------------------------ 
  Line   Exceptions/Handler.php                                            
 ------ ------------------------------------------------------------------ 
  51     Call to an undefined method                                       
         Illuminate\Contracts\Routing\ResponseFactory::unauthenticated().  
  54     Call to an undefined method                                       
         Illuminate\Contracts\Routing\ResponseFactory::unauthenticated().  
  57     Call to an undefined method                                       
         Illuminate\Contracts\Routing\ResponseFactory::notFound().         
  63     Call to an undefined method                                       
         Illuminate\Contracts\Routing\ResponseFactory::tooManyRequests().  
 ------ ------------------------------------------------------------------ 
```

### Laravel code where the issue was found

```php
<?php

namespace App\Providers;

use Illuminate\Support\Facades\Response;
use Illuminate\Support\ServiceProvider;

class ResponseServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap the application services.
     */
    public function boot(): void
    {
        Response::macro('noContent', function () {
            return Response::json('', 204);
        });

        Response::macro('deleted', function () {
            return Response::noContent();
        });

        Response::macro('resetContent', function () {
            return Response::json([], 205);
        });
    }

    /**
     * Register the application services.
     */
    public function register(): void
    {
        //
    }
}
```

**Comment by szepeviktor** [comment]
Thank you for your report.

Could you retry with yesterday's release? https://github.com/nunomaduro/larastan/releases

@nunomaduro Is it really a **Draft** release?

**Comment by canvural** [comment]
@szepeviktor Yeah, it's still a draft. It's not released yet.

But I doubt the new release will fix these macro issues. We have to take a look at it.

**Comment by canvural** [comment]
Hi,

This might be fixed with #570 Can you try with the `master` branch and see if the issue still continues?

**Comment by Nizari** [comment]
The `master` branch fixed this issue :)! Looking forward to `0.5.9`!

**Comment by canvural** [comment]
Thank you for kind words! Glad it fixed the issue.

BTW, the next version will be `0.6` Because there might be some small breaking changes.
